byterun/shared_heap.c

```diff
@@ -220,6 +232,62 @@ static pool* pool_find(struct caml_heap_state* local, sizeclass sz) {
   r = local->avail_pools[sz];
   if (r) return r;
 
+/* Haven't managed to find a pool locally, try the global ones */
+  caml_plat_lock(&pool_freelist.lock);
+  if( pool_freelist.global_avail_pools[sz] ) {
+    r = pool_freelist.global_avail_pools[sz];
+
+    if( r ) {
+      pool_freelist.global_avail_pools[sz] = r->next;
+      r->next = 0;
+      local->avail_pools[sz] = r;
+
+      int free_objs = 0;
```

Comment:
Apart from the assert in debug mode, I can't see next_obj and free_objs actually being used. Should we but this in DEBUG guards (or even remove)?

Reply:
You're right, this is left over from some existing free space calculation code and we should only do it in DEBUG to verify the pool.